benches/divan.rs

```diff
@@ -0,0 +1,103 @@
+fn main() {
+    init_tracing();
+
+    #[cfg(feature = "tracing")]
+    let _span = tracing::span!(tracing::Level::DEBUG, "BENCH EXAMPLES").entered();
+
+    // Run registered benchmarks.
+    divan::main();
+}
+
+#[cfg(feature = "tracing")]
+fn init_tracing() -> impl Drop {
+    use tracing_flame::FlameLayer;
+    use tracing_subscriber::{fmt, prelude::*};
+    let filter_layer = tracing_subscriber::EnvFilter::from_default_env();
+    let fmt_layer = fmt::Layer::default();
+    let (flame_layer, _guard) = FlameLayer::with_file("./tracing.folded").unwrap();
+
+    tracing_subscriber::registry()
+        .with(filter_layer)
+        .with(fmt_layer)
+        .with(flame_layer)
+        .init();
+
+    _guard
+}
+
+#[cfg(not(feature = "tracing"))]
+fn init_tracing() {}
+
+mod examples {
+    #[divan::bench]
+    fn parse_postal(bencher: divan::Bencher) {
+        let input = divan::black_box(include_str!(
+            "../tests/fixtures/postal_address.terminated.input.bnf"
+        ));
+
+        bencher.bench(|| {
+            input.parse::<bnf::Grammar>().unwrap();
+        });
+    }
+
+    #[divan::bench]
+    fn generate_dna(bencher: divan::Bencher) {
+        bencher
+            .with_inputs(|| {
+                let input = "<dna> ::= <base> | <base> <dna>
+            <base> ::= 'A' | 'C' | 'G' | 'T'";
+                let grammar: bnf::Grammar = input.parse().unwrap();
+                grammar
+            })
+            .bench_refs(|grammar| {
+                grammar.generate().unwrap();
+            });
+    }
+
+    #[divan::bench]
+    fn parse_polish_calculator(bencher: divan::Bencher) {
+        let polish_calc_grammar: bnf::Grammar = "<product> ::= <number> | <op> <product> <product>
+            <op> ::= '+' | '-' | '*' | '/'
+            <number> ::= '0' | '1' | '2' | '3' | '4' | '5' | '6' | '7' | '8' | '9'
+        "
+        .parse()
+        .unwrap();
+
+        // use pseudo random for consistent metrics
+        let mut rng: rand::rngs::StdRng = rand::SeedableRng::seed_from_u64(0);
+        let random_walk_count = 100usize;
+        let random_walks: Vec<_> = (0..random_walk_count)
+            .map(|_| polish_calc_grammar.generate_seeded(&mut rng).unwrap())
+            .collect();
+
+        bencher
+            .with_inputs(|| {
+                let mut rng = rng.clone();
+                use rand::seq::SliceRandom;
+                random_walks.choose(&mut rng).unwrap()
+            })
```

Comment:
Likewise this is using the same input for every iteration. It would probably be good to model this after `parse_infinite_nullable_grammar`.